### ISSUE TITLE
Monitoring API is not ready

### ISSUE THREAD

**Issue by gywang-** [rancher/rancher#19665]
*enabled monitor fail*

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.2.1` UI: `v2.2.19` |
|Route|`authenticated.cluster.monitoring.cluster-setting`|

I have two k8s clusters that have monitoring turned on,But one normal one is not normal.

normal clusters:
```shell
kubectl get deploy -n  cattle-prometheus
NAME                                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
exporter-kube-state-cluster-monitoring    1         1         1            1           1d
grafana-cluster-monitoring                1         1         1            1           1d
prometheus-operator-monitoring-operator   1         1         1            1           1d
```
![image](https://user-images.githubusercontent.com/10112569/56256316-b2d67380-60fa-11e9-96b6-abe81e0efeae.png)

not normal clusters, Not started “prometheus-operator-monitoring-operator ” why?
```shell
kubectl get deploy -n  cattle-prometheus
NAME                                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
exporter-kube-state-cluster-monitoring   1         1         1            1           11h
grafana-cluster-monitoring               1         1         1            1           11h
```

![image](https://user-images.githubusercontent.com/10112569/56256295-a6eab180-60fa-11e9-9afc-8905e5a851ba.png)

![image](https://user-images.githubusercontent.com/10112569/56256339-d5688c80-60fa-11e9-8d99-1b628e65cbf6.png)

**Comment by alena1108** [comment]
@gywang- we've recently released Rancher v2.2.2 that had a fix for the issue that is similar to what you see: https://github.com/rancher/rancher/issues/19380. Can you give monitoring a try on 2.2.2?

**Comment by mateuszkwiatkowski** [comment]
@alena1108 we're seeing this issue on Rancher 2.2.2. I've tested in on cluster which was created with rke and then imported into Rancher.

**Comment by loganhz** [comment]
@mateuszkwiatkowski Can you go to System project and provide the screen shorts for the workloads under `cattle-prometheus` namespace?

**Comment by jiaqiluo** [comment]
The bug is reproduced only on a 3-node (all roles on each node) imported RKE cluster. 

Rancher v2.2.2

cluster |  k8s version | enable cluster monitoring
-------| ------ | ------------
RKE - EC2 | v1.13.5  | work
ASK             | v1.12.7|  work
GKE             | v1.12.6-eks-d69f1b |  work
EKS             |  v1.12.6-eks-d69f1b | work
GKE - imported  | v1.11.8-gke.6  |  work
RKE - imported (single node) | v1.13.4 | work
RKE - imported (**3 nodes**) | v1.13.4 | **NOT** work

Steps:
- run Rancher:v2.2.2 container, single installed
- import a 3-node RKE cluster (all roles on each node)
- enable cluster monitoring

Result:
- the workload `exporter-node-cluster-monitoring` cannot work properly
 
<img width="1373" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/56381559-167ba200-61ca-11e9-91a8-32db820d970c.png">

- the log of the pod  `exporter-node-cluster-monitoring-g8nd9`

```
msg="listen tcp 18.222.184.184:9796: bind: cannot assign requested address" source="node_exporter.go:114"
```

**Comment by davidnuzik** [comment]
Version: Rancher v2.2.2
When I use (unsupported) rke v0.1.17 and create a multi-node rke import cluster this works without any issues for me - I can enable monitoring and it works and the exporter-nodes deploy without any issues. Thus, we think this may have to be something to do with RKE.

**Comment by orangedeng** [comment]
After I did `kubectl get nodes -o yaml`, I got
```
  status:
    addresses:
    - address: 18.223.237.40
      type: InternalIP
    - address: 18.223.237.40
      type: Hostname
```

And I get host internal ip is:
```
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc mq state UP group default qlen 1000
    link/ether 02:01:5f:55:2a:4c brd ff:ff:ff:ff:ff:ff
    inet 172.31.7.87/20 brd 172.31.15.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::1:5fff:fe55:2a4c/64 scope link 
       valid_lft forever preferred_lft forever
```

There is no way to let node exporter to know the internal ip. So it can not bind it. I think there is something wrong with the kubelet that assign an external ip to the internal ip field.

**Comment by gywang-** [comment]
> @ gywang-我们最近发布了Rancher v2.2.2，它修复了与你看到的类似的问题：[＃19380](https://github.com/rancher/rancher/issues/19380)。您能否对2.2.2进行监控？
Still have this problem to v2.2.2，My k8s cluster is my own deployment.

The version is as follows:

```shell
kubectl get  nodes
NAME          STATUS    ROLES     AGE       VERSION
10.10.0.199   Ready     <none>    253d      v1.10.5
10.10.0.208   Ready     <none>    253d      v1.10.5
10.10.1.2     Ready     <none>    40d       v1.10.5
```

```shell
kubectl api-versions
admissionregistration.k8s.io/v1beta1
apiextensions.k8s.io/v1beta1
apiregistration.k8s.io/v1
apiregistration.k8s.io/v1beta1
apps/v1
apps/v1beta1
apps/v1beta2
authentication.istio.io/v1alpha1
authentication.k8s.io/v1
authentication.k8s.io/v1beta1
authorization.k8s.io/v1
authorization.k8s.io/v1beta1
autoscaling/v1
autoscaling/v2beta1
batch/v1
batch/v1beta1
certificates.k8s.io/v1beta1
config.istio.io/v1alpha2
events.k8s.io/v1beta1
extensions/v1beta1
monitoring.coreos.com/v1
networking.istio.io/v1alpha3
networking.k8s.io/v1
policy/v1beta1
rbac.authorization.k8s.io/v1
rbac.authorization.k8s.io/v1beta1
storage.k8s.io/v1
storage.k8s.io/v1beta1
v1
```

```shell
kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by gywang-** [comment]
@alena1108
rancher server error logs 

```shell
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-8lwf7/migrate-group-projectalert-workload [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertRuleController p-8lwf7/memory-close-to-resource-limited [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-qncqn/projectalert-workload-alert [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertRuleController p-8lwf7/less-than-half-workload-available [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/high-memmory [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-xwt25/migrate-group-projectalert-workload [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-8lwf7/projectalert-workload-alert [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/high-cpu-load [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertRuleController p-xwt25/less-than-half-workload-available [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertRuleController p-qncqn/less-than-half-workload-available [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/migrate-clusteralert-node-mem [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-qncqn/migrate-group-projectalert-workload [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertRuleController p-qncqn/memory-close-to-resource-limited [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/no-leader [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertRuleController p-xwt25/memory-close-to-resource-limited [project-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/controllermanager-system-service [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-xwt25/pag-x5j2c [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ProjectAlertGroupController p-xwt25/projectalert-workload-alert [project-alert-group-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/migrate-clusteralert-deploment-event [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/deploment-event-alert [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/migrate-clusteralert-controllermanager [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/scheduler-system-service [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/migrate-clusteralert-etcd [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/etcd-system-service [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/db-over-size [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/deployment-event-alert [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/high-number-of-leader-changes [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/migrate-clusteralert-scheduler [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:44 [ERROR] ClusterAlertRuleController c-b8lcl/node-disk-running-full [cluster-alert-rule-controller] failed with : Failed to get service for alertmanager, service "cattle-prometheus/alertmanager-operated" not found
2019/04/21 03:51:48 [ERROR] CatalogController helm-incubator [catalog] failed with : Timeout in HTTP GET to [https://kubernetes-charts-incubator.storage.googleapis.com/index.yaml], did not respond in 30s
```

**Comment by thxCode** [comment]
@gywang- , could you confirm that Is there any error msgs in the `monitoring-operator` App of `System` Project in that failed cluster?

**Comment by gywang-** [comment]
> @gywang- , could you confirm that Is there any error msgs in the `monitoring-operator` App of `System` Project in that failed cluster?

How should I check if there is an error?

**Comment by gywang-** [comment]
[Solved using this method](https://github.com/rancher/rancher/issues/19209#issuecomment-479404791)

**Comment by loganhz** [comment]
@jiaqiluo  Please make sure the internal IP is correct see https://github.com/rancher/rancher/issues/19665#issuecomment-484762494

**Comment by jiaqiluo** [comment]
@loganhz  There is no need to provide the internal IPs when using RKE to bring up a rke cluster, so if the cluster is imported into Rancher, we expect the monitoring works. Are we saying that the user needs to provide the internal ip in the yaml file when creating an RKE cluster for importing it into Rancher?  If this is the case, we should update the document.

**Comment by jiaqiluo** [comment]
The bug fix is validated on Rancher: master  5d74988c

The same steps as my previous comment describe:
- run Rancher container, single installed
- use RKE: v0.1.18 to provision a 3-node RKE cluster (all roles on each node)
- import the cluster into Rancher
- enable/disable cluster monitoring
- enable/disable project monitoring

Result:
- Both cluster and project monitoring work as expected.
